pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java

```diff
@@ -258,4 +259,25 @@ public static CompletionException wrapToCompletionException(Throwable throwable)
             return new CompletionException(throwable);
         }
     }
+
+    /**
+     * Creates a new {@link CompletableFuture} instance catching
```

Comment:
The return type is "void"
Maybe we can reword this like:

"Executes an operation using the supplied Executor and notify failures on the supplied CompletableFuture"